src/components/Attachment/VoiceRecording.tsx

```diff
@@ -0,0 +1,137 @@
+          <div className='str-chat__message-attachment__voice-recording-widget__timer'>
+            {attachment.duration ? (
+              displayDuration(secondsElapsed)
+            ) : (
+              <FileSizeIndicator fileSize={attachment.file_size} maximumFractionDigits={0} />
+            )}
+          </div>
+          <WaveProgressBar progress={progress} seek={seek} waveformData={waveform_data || []} />
+        </div>
+      </div>
+      <div className='str-chat__message-attachment__voice-recording-widget__right-section'>
+        {isPlaying ? (
+          <PlaybackRateButton disabled={!audioRef.current} onClick={increasePlaybackRate}>
+            {playbackRate.toFixed(1)}x
+          </PlaybackRateButton>
+        ) : (
+          <FileIcon big={true} mimeType={mime_type} size={40} version={'2'} />
+        )}
+      </div>
+    </div>
+  );
+};
+
+export type QuotedVoiceRecordingProps<
+  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics
+> = Pick<VoiceRecordingProps<StreamChatGenerics>, 'attachment'>;
+
+export const QuotedVoiceRecording = ({ attachment }: QuotedVoiceRecordingProps) => {
+  const { t } = useTranslationContext();
+  const title = attachment.title || t<string>('Voice message');
+  return (
+    <div className={rootClassName} data-testid='quoted-voice-recording-widget'>
+      <div className='str-chat__message-attachment__voice-recording-widget__metadata'>
+        {title && (
+          <div
+            className='str-chat__message-attachment__voice-recording-widget__title'
+            data-testid='voice-recording-title'
+            title={title}
+          >
+            {title}
+          </div>
+        )}
+        <div className='str-chat__message-attachment__voice-recording-widget__audio-state'>
+          <div className='str-chat__message-attachment__voice-recording-widget__timer'>
+            {attachment.duration ? (
+              displayDuration(attachment.duration)
+            ) : (
+              <FileSizeIndicator fileSize={attachment.file_size} maximumFractionDigits={0} />
+            )}
+          </div>
+        </div>
+      </div>
+      <FileIcon big={true} mimeType={attachment.mime_type} size={34} version={'2'} />
+    </div>
+  );
+};
+
+export type VoiceRecordingProps<
+  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics
+> = {
+  /** The attachment object from the message's attachment list. */
+  attachment: Attachment<StreamChatGenerics>;
+  /** A boolean flag to signal whether the attachment will be rendered inside the quoted reply. */
+  isQuoted?: boolean;
+};
+
+const UnMemoizedVoiceRecording = ({ attachment, isQuoted }: VoiceRecordingProps) =>
+  isQuoted ? (
+    <QuotedVoiceRecording attachment={attachment} />
+  ) : (
+    <VoiceRecordingPlayer attachment={attachment} />
+  );
+
+export const VoiceRecording = React.memo(
+  UnMemoizedVoiceRecording,
+) as typeof UnMemoizedVoiceRecording;
```

Comment:
Question... do we have to memoize the new components? We know this causes problems in the long run.

Reply:
Yes, will remove it